src/ToEnum.php

```diff
@@ -21,53 +25,39 @@
  */
 final class ToEnum implements FilterInterface
 {
-    /**
-     * @var class-string<UnitEnum>|null
-     */
+    /** @var class-string<UnitEnum>|null */
     private ?string $enumClass = null;
 
     /**
-     * @param class-string<UnitEnum>|Traversable|Options $enumOrOptions
+     * @param Traversable|class-string<UnitEnum>|Options $enumOrOptions
      */
     public function __construct($enumOrOptions)
     {
         if ($enumOrOptions instanceof Traversable) {
+            /** @var Options $enumOrOptions */
             $enumOrOptions = ArrayUtils::iteratorToArray($enumOrOptions);
         }
 
         if (
             is_array($enumOrOptions) &&
             isset($enumOrOptions['enum'])
         ) {
-            $this->setEnum($enumOrOptions['enum']);
-
-            return;
+            $this->enumClass = $enumOrOptions['enum'];
         }
 
         if (is_string($enumOrOptions)) {
-            $this->setEnum($enumOrOptions);
+            $this->enumClass = $enumOrOptions;
         }
     }
 
-    /**
-     * @param class-string<UnitEnum> $enum
-     */
-    protected function setEnum(string $enum): self
-    {
-        $this->enumClass = $enum;
-
-        return $this;
-    }
-
     /**
      * Defined by Laminas\Filter\FilterInterface
      *
-     * Returns an enum representation of $value or null
+     * Returns an enum representation of $value if matching.
      *
      * @param  mixed $value
-     * @return UnitEnum|null
      */
```

Comment:
A more precise return type is still useful here

```suggestion
     * @template TInput of mixed
     * @return UnitEnum|TInput
     */
```

Or just:

```suggestion
     * @return UnitEnum|mixed
     */
```

Reply:
Hmm Codesniffer does not like it

Reply:
Probably infers mixed|mixed because on 8.0 `UnitEnum` doesn't exist (maybe). `@psalm-return` should make phpcs happy here, but psalm will still complain about `UndefinedDocblockClass`